### PR TITLE
feat(feed): improve search bar testing and refactor content logic

### LINKED ISSUE #488: Improve Feed Screen Search Functionality and Code Structure
### Description
The feed screen's search functionality needs comprehensive testing and the content display logic requires refactoring to improve maintainability and readability.

### Acceptance Criteria
- [ ] Add comprehensive UI tests for search bar functionality covering all result types
- [ ] Test mixed search results (users, events, organizations)
- [ ] Test individual result type scenarios
- [ ] Verify loading states work correctly in both search and feed modes
- [ ] Test event list loading indicators
- [ ] Refactor FeedContentStateSwitcher into smaller, focused composables
- [ ] Improve code organization and maintainability
- [ ] Ensure all existing functionality remains unchanged
- [ ] All tests must pass

### PR DESCRIPTION
**Pull request by M-e-d-A**
# Description:
## Purpose of the change
This PR enhances the feed screen's search functionality with comprehensive testing and refactors the content display logic for better maintainability, as defined in issue #488.

It includes:
* 8 new comprehensive UI tests for search bar functionality covering mixed results, individual result types, loading states, and edge cases
* Complete refactoring of `FeedContentStateSwitcher` into focused composables: `SearchContent`, `FeedContent`, `SearchResultsContent`, `EventListContent`, and `CenteredLoadingState`
* Improved code organization and readability while maintaining all existing functionality
* Enhanced test coverage for feed screen search and display logic

## Related issues
Closes #488

## How to test
1. Check out the branch.
2. Run the tests in `FeedScreenSearchBarTest.kt` via Android Studio or `./gradlew test`.
3. Verify all 8 new tests and all existing tests pass.
4. Build and run the app to verify the feed screen functions correctly:
   - Test search with different query types
   - Verify all search result types display correctly
   - Check loading states during search
   - Ensure event feed displays properly
5. Review the refactored code structure in `FeedScreen.kt` for improved maintainability.

**Note:** An LLM was used for the sole and exclusive purpose of improving the clarity and formatting of the PR description.

The refactoring establishes a cleaner separation of concerns, making the feed screen codebase more maintainable while the comprehensive tests ensure robust search functionality.

### CODE REVIEW

File: app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt [hunk 1]
```diff
@@ -287,113 +289,192 @@ private fun FeedContentStateSwitcher(
 ) {
   LazyColumn(
       modifier = Modifier.fillMaxSize().testTag(FeedScreenTestTags.EVENT_LIST),
+      state = listState,
```
[the remark this answers is not in the document]
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt]: Thanks 👍

File: app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt [hunk 2]
```diff
@@ -287,113 +289,192 @@ private fun FeedContentStateSwitcher(
...
+    uiState.events.isNotEmpty() -> {
+      EventListContent(
+          events = uiState.events,
+          isLoadingMore = uiState.isLoading && !uiState.isRefreshing,
+          eventCardViewModel = eventCardViewModel,
+          likedEvents = likedEvents,
+          onNavigateToEvent = onNavigateToEvent)
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt]: Appreciate it!

File: app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt [hunk 3]
```diff
@@ -287,113 +289,192 @@ private fun FeedContentStateSwitcher(
...
-                        onClick = {
-                          globalSearchItemClickListener?.onItemClick(
-                              GlobalSearchItemClick.OrganizationClick(org.id))
-                        },
-                        modifier = Modifier.testTag(getTestTagForSearchOrg(org.id)))
-                  }
-                }
-              }
+/** Displays the actual search results (users, events, organizations). */
+private fun LazyListScope.SearchResultsContent(
```
[the remark this answers is not in the document]
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt]: Appreciate it 👍

File: app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt [hunk 4]
```diff
@@ -287,113 +289,192 @@ private fun FeedContentStateSwitcher(
...
+          isLoadingMore = uiState.isLoading && !uiState.isRefreshing,
+          eventCardViewModel = eventCardViewModel,
+          likedEvents = likedEvents,
+          onNavigateToEvent = onNavigateToEvent)
+    }
+  }
+}
+
+/** Displays the list of events with optional loading indicator at the bottom. */
+private fun LazyListScope.EventListContent(
```
[the remark this answers is not in the document]
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt]: Thanks!
